src/Models/Concerns/HasPrefixedId.php

```diff
@@ -32,6 +33,15 @@ public static function findByPrefixedId(string $prefixedId): ?Model
         return static::firstWhere($attributeName, $prefixedId);
     }
 
+    public static function findByPrefixedIdOrFail(string $prefixedId): Model
+    {
+        if (! is_null($model = static::findByPrefixedId($prefixedId))) {
```

Comment:
Reverse this condition: throw the exception in the if, return the model at the bottom of the function

Reply:
@freekmurze I have reversed the condition.